### ISSUE TITLE
Memory leak

### ISSUE THREAD

**Issue by jqn** [chagasaway/react-native-fading-slides#5]
Hi, thanks for this awesome component. I just wanted to mention I've been receiving this error on android devices when the component is unmounted 
```
index.android.bundle?platform=android&dev=true&hot=false&minify=false:11798 Warning: Can only update a mounted or mounting component. This usually means you called setState, replaceState, or forceUpdate on an unmounted component. This is a no-op.

Please check the code for the FadingSlides component.
```

I see you are using timer mixing so I'm not sure why this would happen.
I'm on React Native version 0.44.2

**Comment by chagasaway** [comment]
Hey @jqn, thanks for the issue report.

I will work on it this weekend to release a new version of it, it's really outdated right now.

**Comment by chagasaway** [comment]
Hey @jqn, I replaced the **TimerMixin** and **TweenState** with **Animated API**.

Checkout the new **0.2.0** version published on NPM.

Please let me know here if the issue has been solved with this update.

**Comment by jqn** [comment]
Will do. Thanks for updating this @chagasaway

**Comment by jqn** [comment]
@chagasaway thanks for the quick updates. I tested the updated library and I'm still getting the same error. I'll try and see if I can help figure out how to fix it.

**Comment by chagasaway** [comment]
@jqn, this is strange, since i'm using the AnimatedApi now, could you provide a working project with this issue? Maybe it will be easier to find the memory leak.

**Comment by jqn** [comment]
sure I'll work on that. For now I can tell you I'm using react navigation connected with redux. The error shows up on log out from the app. During this event I unmount all the screens and reset the app to it's initial state  then I received the error twice after that. I'm assuming this function is called twice after that 

```
  _changeSlide = () => {
    let index = this.state.currentIndex + 1;
    index = index < this.props.slides.length ? index : 0;
    this.setState({ currentIndex: index }, this._show);
  };
```
and `this.setState({ currentIndex: index }, this._show);` triggers the error.
This is in android only by the way.

**Comment by chagasaway** [comment]
Hey @jqn, I am closing this since your PR #8 merged and the 0.3.1 is released on NPM.

**Comment by jqn** [comment]
Awesome, thanks again @chagasaway
